src/BuildScriptGenerator/DotNetCore/DotnetCoreScriptGeneratorOptions.cs

```diff
@@ -16,5 +16,7 @@ public class DotNetCoreScriptGeneratorOptions
         /// Gets or sets the MSBuild configuration that needs to be used when doing 'dotnet build'.
         /// </summary>
         public string MSBuildConfiguration { get; set; }
+
+        public string DotNetCoreRuntimeVersion { get; set; }
```

Comment:
Wondering why we are moving this `ResolvedPlatformRuntimeVersion` from Context into Option?

Reply:
We have been doing this before my change:
1. Run detect on all platforms by calling the `Detect` on `IProgammingPlatform`
2. Once we get the detection result, we invoke the `SetVersion` on `IProgrammingPlatform` to enable the platform which detected to set the version on context. We did this way because we wanted the platform to set the particular, ex: `ResolvedDotNetCoreVersion`, `ResolvedPythonVersion` property which the platform knows about
4. We then call the platform's `GenetateBashScriptSnippet` to generate script, which in turn used the version set on the on the context to generate script.

Based on the above flow, you should notice that this flow is complicated and could be simplified. With this PR we will be passing in the detected result directly to the method being called, which is `GenetateBashScriptSnippet`

Reply:
Makes sense.. that's good way to simplify the flow.